documentation/STRUCTURE.md

```diff
@@ -0,0 +1,195 @@
+# Documentation
+
+For the purpose of this proposal document we consider all of the following as documentation:
+
+- documentation in our code within the Fairlearn repository which is currently also hosted on readthedocs
+- Jupyter notebooks
+- project website
+- blog posts
+
+Throughout the document we compare with the documentation of other popular projects such as 
+
+- [scikit-learn](https://scikit-learn.org/)
+- [pandas](https://pandas.pydata.org/)
+
+## Goals
+
+The documentation should be
+
+- discoverable, ideally in a single place as opposed to multiple
+- clear
+- concise when describing individual pieces of functionality
+- detailed when describing entire application scenarios, e.g. in the form of case studies
+- available for the latest version, but if possible also for past versions ([example](https://scikit-learn.org/dev/versions.html))
+- maintainable: it should be simple for maintainers to update/validate
+- without ads (readthedocs always has ads that are shown alongside our documentation)
+
+## Proposal
+
+Like for most projects the website, [fairlearn.org](http://fairlearn.org) will be the central place to look for documentation.
+From there visitors have various paths to explore content depending on what they are looking for, as detailed in the following subsections.
+
+### Homepage
+
+```
+website
+|--- Quickstart
+|--- API reference
+|--- User guide
+|--- Example Notebooks
+|--- Contributor guide
+|--- Community
+```
+
+### Quickstart
+
+This page provides information on
+- Installation
+- brief introduction/framing of fairness in ML
+  - including basic terminology (perhaps link to more comprehensive section on a different page)
+- walk-through
+  - load data
+  - mitigate disparity of an estimator
+  - evaluate a few metrics
+  - run the dashboard
+- links showing where to go next, e.g. links to section of the user guide
+
+Installation should cover various platforms, which should be very straightforward. Any reoccurring patterns in reported issues should be listed, as well as how to troubleshoot them. It may end up similar to [this guide](https://scikit-learn.org/dev/install.html)).
+
+Example:
+
+- [pandas](https://pandas.pydata.org/getting_started.html)
+- [scikit-learn](https://scikit-learn.org/dev/getting_started.html)
+
+This should be very similar to what we currently have in our README, so a lot of the content won't be entirely new.
+
+### API reference
+
+This is simply the generated documentation from our code. Currently we host this in readthedocs, but we want to include it on our webpage. A good example for this is [scikit-learn](https://scikit-learn.org/dev/modules/classes.html)
+
+### User guide
+
+The user guide explains all parts of Fairlearn by providing context that wouldn't fit into the code documentation such as mathematical derivations, but without using application-specific context (as we'd find it in the "case studies"). The guides are grouped by topic, e.g.
```

Comment:
If someone who cares about "fairness is fundamentally a sociotechnical problem" ends up here, what might they find here that would keep them engaged and interested in this project?

Reply:
Depends! Does that person know about fairness (and why it's complicated) already, and are they looking to understand how it's related to AI systems?

If they're interested in specific examples of how this may be approached in practice with a concrete example then the example notebooks section is probably more interesting.

The user guide is about educating people about fairness, why it's complicated, links to research about it in case they want further information, and explaining what Fairlearn does in this space (which is a small part that focuses on only a small slice). Looking at the overview below you probably don't get a good idea about proportions because the first section is only one line, whereas everything else looks like much more. In reality that may not turn out like that. If we can get quality content in there that fits into the overall picture I'm more than happy to add it. I'm writing "if" because it's work and not as simple as writing the other sections about how to use some of our code. I'm currently iterating on this and everything that's there is basically migrated from the existing markdown pages we had in the repository. Any feedback is welcome, and feel free to address problems with PRs. https://fairlearn.github.io/user_guide/fairness_in_machine_learning.html

So to circle back to your question: Once this is in good shape there will be content there explaining why fairness is a sociotechnical problem, what that means, and how that affects AI systems. I imagine that a lot of the content will be related to Hanna and Jenn's [webinar on Fairness in ML](https://note.microsoft.com/MSR-Webinar-Machine-Learning-and-Fairness-Registration-LIVE.html) (requires sign-up, but is free). The way they frame the problem allows for various ways to utilize Fairlearn, but obviously there's more to the problem than measuring and mitigating unfairness. I assume that's what you're trying to get at, and I'm hoping we can make that clear. Any suggestions are obviously welcome!